ui/src/app/core/settings.service.ts

```diff
@@ -14,7 +15,8 @@ type ProjectSettings = {
   // set to '' for non-project-based job manager
   projectId: string,
   displayColumns: string[],
-  pageSize: number
+  pageSize: number,
+  hideArchived: boolean
```

Comment:
This should be more generic. Would it be difficult to generalize it to something like:
```
{
  ...
  filters: Filter[] 
}

type Filter = {
  key: string
  value: any
}
```

That way we don't have to special case `hideArchived` in the UI, we can just store all current filters in settings and when you return back to the JM for that project the same filters will automatically be in place. I realize this is a much bigger ask, but I'm pretty against hard coding things like this in the UI since it sends us down the wrong path of adding code specific to a given backend. Ideally this would either be implemented in a general way or not stored in local storage at first.

From a product standpoint, why is this the only filter we want to store? It seems like either all your previous filters should show up, or none of them... @ruchim

Reply:
@bfcrampton for the associated user story, see issue #316

Reply:
@ruchim How unique is the `hideArchived` scenario?  Would users like to have all the attributes of the last query they used show up in new windows or the next time they use JMUI?

Reply:
@bfcrampton Got a clarification offline – new requirement is to retain all the query filters in local storage.  Will notify you when it's ready for review again